slice/internal/webhooks/jobset_webhook_test.go

```diff
@@ -21,105 +21,193 @@ import (
-								SetAnnotation(PodSetRequiredTopologyAnnotation, AnnotationValueTBD).
-								SetAnnotation(PodSetSliceRequiredTopologyAnnotation, AnnotationValueTBD).
-								SetAnnotation(PodSetSliceSizeAnnotation, AnnotationValueTBD).
-								Obj(),
-						},
-					},
-				},
-				ObjectMeta: ctrl.ObjectMeta{
-					Labels: map[string]string{
-						kueueconstants.QueueLabel: "local-queue",
-					},
-					Namespace: metav1.NamespaceDefault,
-				},
-			},
+		"no tpu topology annotation": {
+			jobSet: testingjobjobset.MakeJobSet(baseJobSetName, utils.DefaultNamespace).
+				Queue("queue-name").
+				ReplicatedJobs(testingjobjobset.ReplicatedJobRequirements{
+					Name:        "rj1",
+					Parallelism: 12,
+					NodeSelector: map[string]string{
+						TPUAcceleratorLabel: "tpu-v7x",
+					},
+				}).
+				Obj(),
+			wantJobSet: testingjobjobset.MakeJobSet(baseJobSetName, utils.DefaultNamespace).
+				Queue("queue-name").
+				ReplicatedJobs(testingjobjobset.ReplicatedJobRequirements{
+					Name:        "rj1",
+					Parallelism: 12,
+					NodeSelector: map[string]string{
+						TPUAcceleratorLabel: "tpu-v7x",
+					},
+				}).
+				Obj(),
+		},
+		"no tpu accelerator node selector label": {
+			jobSet: testingjobjobset.MakeJobSet(baseJobSetName, utils.DefaultNamespace).
+				Queue("queue-name").
+				ReplicatedJobs(testingjobjobset.ReplicatedJobRequirements{
+					Name:        "rj1",
+					Parallelism: 12,
+					PodAnnotations: map[string]string{
+						TPUTopologyAnnotation: "4x4x12",
+					},
+				}).
+				Obj(),
+			wantJobSet: testingjobjobset.MakeJobSet(baseJobSetName, utils.DefaultNamespace).
+				Queue("queue-name").
+				ReplicatedJobs(testingjobjobset.ReplicatedJobRequirements{
+					Name:        "rj1",
+					Parallelism: 12,
+					PodAnnotations: map[string]string{
+						TPUTopologyAnnotation: "4x4x12",
+					},
+				}).
+				Obj(),
+		},
+		"should set default values": {
+			jobSet: testingjobjobset.MakeJobSet(baseJobSetName, utils.DefaultNamespace).
+				Queue("queue-name").
+				ReplicatedJobs(testingjobjobset.ReplicatedJobRequirements{
+					Name:        "rj1",
+					Parallelism: 12,
+					PodAnnotations: map[string]string{
+						TPUTopologyAnnotation: "4x4x12",
+					},
+					NodeSelector: map[string]string{
+						TPUAcceleratorLabel: "tpu-v7x",
+					},
+				}).
+				Obj(),
+			wantJobSet: testingjobjobset.MakeJobSet(baseJobSetName, utils.DefaultNamespace).
+				Queue("queue-name").
+				ReplicatedJobs(testingjobjobset.ReplicatedJobRequirements{
+					Name:        "rj1",
+					Parallelism: 12,
+					PodAnnotations: map[string]string{
+						TPUTopologyAnnotation:                            "4x4x12",
+						kueuealpha.PodSetRequiredTopologyAnnotation:      TPUBlockLabel,
+						kueuealpha.PodSetSliceRequiredTopologyAnnotation: TPUSubBlockLabel,
+						kueuealpha.PodSetSliceSizeAnnotation:             "4",
+					},
+					NodeSelector: map[string]string{
+						TPUAcceleratorLabel: "tpu-v7x",
+					},
+				}).
+				Obj(),
+		},
```

Comment:
Can we run the following cases? Perhaps a new structured test, so that these values are easy to specify
4x4x4, parallelism: 16 => 16 (expected size)
4x4x4, parallelism: 64 => 64
4x4x12, parallelism: 48 => 16
4x4x12, parallelism: 96 => 32